pwndbg/heap/ptmalloc.py

```diff
@@ -172,8 +172,9 @@ def mp(self):
 
     @property
     def global_max_fast(self):
-        addr = pwndbg.symbol.address('global_max_fast')
-        return pwndbg.memory.u(addr)
+        # global_max_fast definition:
+        #       https://sourceware.org/git/?p=glibc.git;a=blob;f=malloc/malloc.c;h=96149549758dd424f5c08bed3b7ed1259d5d5664;hb=HEAD#l1630
+        return pwndbg.symbol.address('global_max_fast')
```

Comment:
Btw why do we remove `pwndbg.memory.u` from here and not use it anywhere?

Reply:
Can we put some explanation in the git commit log?

Reply:
`global_max_fast` symbol contains the value

Reply:
The commit message if already clear I think

Reply:
@Segflow I think you found a bug in `pwndbg.symbol.address` here, as its function would be to return the actual address of the symbol, instead of its value. And looking at its other call sites, it does just that. The problem seems to be with intergral symbols (which `global_max_fast` is) because the cast to int [here][0] returns the actual value in the symbol (let's say 0x80 for global_max_fast) instead of the address, but the code fails and falls over to the following lines for other types, where it resolves their addresses properly. I believe that the proper fix would be to replace `return int(symbol_obj)` with `return int(symbol_obj.value().address)` in `pwndbg.symbol.address` instead of working around the issue in the `global_max_fast` function because while currently this is the only instance I could find where we call `pwndbg.symbol.address` with an integral type, there might be others in the future.

See the examples below:

```
pwndbg> pi hex(int(gdb.lookup_symbol('main_arena')[0]))
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/media/SSD2/dev/tools/pwndbg/pwndbg/inthook.py", line 56, in __new__
    return _int(_int(value, *a, **kw))
gdb.error: Cannot convert value to long.
Error while executing Python code.
pwndbg> pi hex(int(gdb.lookup_symbol('main_arena')[0].value().address))
'0x7ffff396db20'

pwndbg> pi hex(int(gdb.lookup_symbol('global_max_fast')[0]))
'0x80'
pwndbg> pi hex(int(gdb.lookup_symbol('global_max_fast')[0].value().address))
'0x7ffff396f7f8'
pwndbg> 
```
[0]: https://github.com/pwndbg/pwndbg/blob/dev/pwndbg/symbol.py#L208

Reply:
true! should be fixed now.